### ISSUE TITLE
Do We Need Help?

### ISSUE THREAD

**Issue by MattColville** [AdventureLookup/AdventureLookup#178]
Now that the site is live, we're getting a ton of feature requests and I don't want to overload @cmfcmf and @harkenn.

I'm going to make a video probably this week, covering hitting 100k subscribers and a basic State of the Nation kind of thing, it would be a good place to ask for help with adventure lookup.

A: Is this a good idea.

B: What should I tell people? What experience does someone need to have to be a good fit to help us?

C: Where do I direct them? Here?

**Comment by evansche** [comment]
And related: where do you want to direct functionality/feature/bug requests versus data/cleanup requests and discussions? I'm one of the new curators, but I have a question about standards for title formatting I'd like to post to the admins and other curators (the deciders), but I don't know if to use Reddit, or here, or? Is there a Discord chat for some of this?

With success comes work. Congratulations on building and launching something in which a TON of folks find value and engagement. That's something!

**Comment by MattColville** [comment]
I think the subreddit is probably the best place for Standards and Practices when it comes to curation.

The discord is useful for asking questions live, but I think the answers will all be about the things said on the subreddit.

This is for feature requests and bug reports.

**Comment by harkenn** [comment]
I think it's a fine idea. It certainly can't do any harm since we approve all the code that goes in. As for qualifications, we're using php/Symfony 3, sass and npm. We're technically using more but not to any significant degree. 

If they want to contribute, they need only to "fork the repository and when your work is complete, submit a pull request against the official repository, dev branch. Once approved, the pull request will be merged in."

**Comment by cmfcmf** [comment]
> A: Is this a good idea.

Yes. I second @harkenn's comment. More developers would be great, because I definitely won't be able to continue putting as much time into the development of AdventureLookup as I do at the moment.

**Comment by marlinpierce** [comment]
Where are the feature requests collected?  They need to be reviewed and approved.  (Sometimes the answer will be "No, won't fix".  Where would a developer find answers to questions about the feature requests?

I forked and cloned locally, got the setup working in vagrant.  I looked through issues here, and it's hard to glean what should be worked on.  Some issues are half baked, with no end decision on how it should work.  That can be fine, but where can someone get your attention to finalize the design and co-ordinate who is working on a given issue?  Basically, I want to know "How do I ask about details for the issues?"

So Matt, I can confirm you should direct developers here, and the trail is marked from there.

**Comment by marlinpierce** [comment]
Also, how should one ask questions like, "How does it work now?", "What is the database schema?"

**Comment by harkenn** [comment]
@marlinpierce so a few things, if there is doubt on direction or clarity, definitely use github's @ to tag people so that they get a notification.

Deciding what to work on, or if a feature isn't desirable, is a subjective process. Not always do we all agree on the importance, and not always to we think the application should go in the same way as Matt does. Thus far that's all been remediated by Matt's diligence to following up on issues here on GH and also through discussion on each github issue.

> How do I ask about details for the issues?
The standard for open source projects that don't have a formal development approval process is to pick issues that look refined enough to support a specific implementation. If you choose poorly or execute the development poorly, you'll be wasting your own time. Assign yourself to things you're working on and expect to finish soon (a few days? less than a week? its annoying to see stuff that was assigned to someone 2 weeks ago and still isn't finished unless its just a really huge feature).

> How does it work now?
For most things, the answer is that if you're unsure you should get to that specific area in the application and test the existing functionality now. For other situations, like people building new functionality that you need to ask "how does it work?" you just ask them "how does it work" in the relevant issue, or on discord, or wherever makes the most sense.

> What is the database schema?
You can see this yourself, if you check the configuration files for the symfony application, you'll see that you're running your own local database inside the VM. After running migrations up you can connect to the mysql server and look at the schema yourself.
This can be done either via `vagrant ssh` and then connecting via mysql command line, but you'll be limited to a CLI version of working in the database. If you'd rather use something like workbench or datagrid, you can connect to the mysql server via ssh tunneling. I was able to get it working and at some point will edit the readme with instructions so that others who are less familiar w/ the concept can do it too.

**Comment by marlinpierce** [comment]
> You can see this yourself, if you check the configuration files for the symfony application, you'll see that you're running your own local database inside the VM. After running migrations up you can connect to the mysql server and look at the schema yourself.

Does symfony have anything like the ruby on rails schema.rb file where all the tables and all their fields are listed, so you don't have to go to each table and do a desc?

**Comment by marlinpierce** [comment]
I gotta say all that "work at your own risk, and we might throw away your effort" is really lousy and is treating people volunteering their effort like shit.

**Comment by etchalon** [comment]
@marlinpierce For table descriptions, just use a graphic MySQL client. There are many very, very fine ones. I personally use https://www.sequelpro.com on OS X.

In terms of throwing away the effort people contribute, that's just the nature of a fluid, open source project. Your options are to either to reach out to the group and develop a clear consensus from the primary maintainers on exactly what it is you'll be doing, or to just wing it, at the risk that the consensus of the maintainers will be either that the work isn't good enough to use, isn't the way the feature should be implemented, or that the implementation is simply not something the project wants.

**Comment by marlinpierce** [comment]
>  Your options are to either to reach out to the group and develop a clear consensus from the primary maintainers on exactly what it is you'll be doing

That was my question.  How?

Also, it wasn't about the answer to what the schema is, but how do I ask a technical question?

**Comment by harkenn** [comment]
To clarify, you're not going to engage with us, ask for specific advice, follow our advice, and then have your PR ignored or declined. We might provide feedback to shore up issues that we see, but we're not going to decline it outright without some really, really good reasons.

No one is looking to waste anyone's time, particularly if you're making a good faith effort to help.

It depends on the nature of the technical question. If its specific to a symfony, doctrine, or a tool we're using, we can help but only so far before we're probably going to direct you towards the official resources. If it has to do with AdventureLookup, ask on the issue pertinent to the question or create a new issue if necessary.

**Comment by marlinpierce** [comment]
Ask where?  On the issue?  Email you?  Email Matt?  Start a sub-reddit?

Sure, sometimes it's clear to use the issue, as for detailed design.  You keep saying just ask, but I don't know where the forum is to ask.

**Comment by harkenn** [comment]
The last sentence of my last response has literally the answer you're asking for.

**Comment by marlinpierce** [comment]
Hey, I'd like to apologize, because I realized I have been obnoxious.  I realize you have been helpful and accommodating, even though I have found it frustrating finding an appropriate way to communicate to avoid stepping on other people's toes.

An example was when I set up the vagrant image and had a problem with elasticsearch.  It turned out that a quick email exchange was very helpful.  It helped to find out that the vagrant image should have been complete and not require a fixup, so try starting over.

And what's the status of issue #21?  Do we have an ETA?  I don't think it's best to have people entering comments in the issue every week asking if the schedule is on track.

And of course, this whole discourse on this issue is way out of control, for issue comments, even though the process needed to be worked through.

**Comment by harkenn** [comment]
If something like the vagrant image and elastic search happens, opening a new issue is definitely the best way. That way when others have the same problem in the future there is a document we can point them to and a document for search engines to cache.

I'll post on 21 my thoughts.
